core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -416,20 +426,20 @@ object Declaration {
     (Indy.parseIndent *> parser).mapF(Padding.parser(_))
 
   // This is something we check after variables
-  private val bindingOp: Indy[(Pattern.Parsed, Region) => Binding] = {
-    BindingStatement.bindingParser[Pattern.Parsed, Padding[Declaration]](parser <* Indy.lift(toEOL), restP)
+  private lazy val bindingOp: Indy[(Pattern.Parsed, Region) => Binding] = {
+    BindingStatement.bindingParser[Pattern.Parsed, Padding[Declaration]](nonBindingParser <* Indy.lift(toEOL), restP)
```

Comment:
this is reaching all the way to the end. This isn't good. We should stop hiding these loops and make them explicit.